### PR TITLE
[plugin.video.retrospect] v5.0.1

### PR DESCRIPTION
**Pull request by basrieter**
### Description

A small update to fix some stuff:

- Fixed: some versioning stuff for channels and fixed the naming problem of revision and build, which were reversed.
- Fixed: video items always showed channel settings in context menu even if a channel has no settings
- Added: Breadcrumbs for current listing
- Language corrections and additions for Swedish language
- Fixed: show some more information when logging on to a channel fails
- Added: if no username was configured, log this as a warning
- Added: allow playback of NPO Plus content if you are logged in with an NPO Plus account
- Added: option to configure the use of the InputStream Adaptive add-on for RTL XL
- Bumped the version of the script.module.inputstreamhelper to 0.3.5 to solve the Widevine issue
- Added: Install Widevine using the InputStream Helper for streams that require it

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.retrospect/resources/libs/chn_class.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def __init__(self, channel_info):
         self.path = channel_info.path
         self.version = channel_info.version
         self.adaptiveAddonSelectable = channel_info.adaptiveAddonSelectable
-        self.hasSettings = channel_info.settings is not None
+        self.hasSettings = channel_info.settings is not None and len(channel_info.settings) > 0
```
Comment (enen92) [inline plugin.video.retrospect/resources/libs/chn_class.py]: why not just `self.hasSettings = bool(channel_info.settings)` ?

File: plugin.video.retrospect/resources/libs/streams/mpd.py [hunk 2]
```diff
@@ -46,6 +46,17 @@ def set_input_stream_addon_input(strm, proxy=None, headers=None,
 
         """
 
+        if license_key is not None:
```
Comment (enen92) [inline plugin.video.retrospect/resources/libs/streams/mpd.py]: Same here, I'm not a fan of `is not None` in places where you can simply do `if something:`

File: plugin.video.retrospect/resources/libs/chn_class.py (hunk 1 above)
Comment (basrieter) [reply plugin.video.retrospect/resources/libs/chn_class.py]: Never know that was possible :)

File: plugin.video.retrospect/resources/libs/streams/mpd.py (hunk 2 above)
Comment (basrieter) [reply plugin.video.retrospect/resources/libs/streams/mpd.py]: I understand. Is it OK if I pick with each piece of code I change for a next release. That way the code will increasingly become nicer.

**Pull request merged**